packages/patterns/test/qp-on-pattern.test.js

```diff
@@ -0,0 +1,38 @@
+import test from '@endo/ses-ava/prepare-endo.js';
+
+import { qp } from '@endo/marshal';
+import { M } from '../src/patterns/patternMatchers.js';
+
+// This is a continuation of a test case in marshal-justin.test.js,
+// but is placed here because it uses patterns which don't exist at
+// the `@endo/marshal` layer.
+//
+// This was the original example in a slack thread looking for a more
+// legible way to render patterns. Patterns are Passables, and all
+// Passables can kinda be rendered in Justin. Though Remotables and
+// promises will be rendered only in terms of slot numbers.
+test('qp on a pattern', t => {
```

Comment:
this is another case where I wish for updoc/doctest, so that the same example can serve as an executable test and documentation.